src/storage/session_store.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+
+This program is licensed to you under the Apache License Version 2.0,
+and you may not use this file except in compliance with the Apache License Version 2.0.
+You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the Apache License Version 2.0 is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+*/
+
+#ifndef SESSION_STORE_H
+#define SESSION_STORE_H
+
+#include "../../include/json.hpp"
+#include <list>
+
+using std::list;
+using json = nlohmann::json;
+
+namespace snowplow {
+/**
+ * @brief Storage interface used by the ClientSession to store and access sessions.
+ *
+ * You may implement your own storage or make use of the `SqliteStorage` provided by the tracker.
+ */
+struct SessionStore {
+  /**
+   * @brief Select the current session into the list – there should be at most 1.
+   * 
+   * @param session_list Output list of sessions (of size 0 or 1)
+   */
+  virtual void get_session(list<json>* session_list) = 0;
```

Comment:
That's a good point! I was thinking the same thing but didn't want to change the Storage API. On the other hand, now is a good time to change it since we are letting users bring their own storage. I changed it to return a pointer to the JSON session.